src/Build/BackEnd/Components/Communications/NodeLauncher.cs

```diff
@@ -191,15 +240,78 @@ private Process StartInternal(string msbuildLocation, string commandLineArgs)
             }
         }
 
+        private static string CreateTransportName(int nodeId)
+        {
+            string transportName = $"msbuild-worker-{nodeId}-for-{Process.GetCurrentProcess().Id}-{DateTime.Now:yyyyMMdd_HHmmss}.socket";
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return transportName;
+            }
+            else
+            {
+                return Path.Combine(Path.GetTempPath(), transportName);
+            }
+        }
+
+        private ProcessStartInfo AddDiagnosticsTracing(ProcessStartInfo startInfo, string transportName)
+        {
+            // Update the environment for the child process to use that pipe name
+            startInfo.EnvironmentVariables["DOTNET_DiagnosticPorts"] = transportName;
+            return startInfo;
+        }
+
+#if !NETFRAMEWORK
+        private void SetUpEventPipeSession(Process process, string transportName)
+        {
+            // * create a list of providers to collect metrics for
+            List<EventPipeProvider> providers = [
+                new EventPipeProvider("Microsoft.Build", System.Diagnostics.Tracing.EventLevel.Verbose)
+            ];
+
+            // * connect a diagnostics session to the child with those providers on our predefined pipe
+            var client = new DiagnosticsClient(process.Id);
+
+            // reduce the default buffer size from 256MB because we will have several child nodes
+            EventPipeSession session = client.StartEventPipeSession(providers, requestRundown: false, circularBufferMB: 32);
+
+            // * start reading events from that session and forwarding them to our own event source
+            // kick the runtime to start the child node
+            // now process the events _somehow_
+            client.ResumeRuntime();
```

Comment:
node spawning seems to be stuck still - I wonder if this bit is setup up correctly?